utils/remote-run

```diff
@@ -123,26 +124,109 @@ class RemoteCommandRunner(CommandRunner):
                 [self.remote_host, '--'] +
                 [quote(arg) for arg in command])
 
-    def sftp_invocation(self):
-        return (['/usr/bin/sftp', '-b', '-', '-q', '-r'] +
-                self.common_options(port_flag='-P') +
-                [self.remote_host])
+    def rsync_invocation(self, source, dest):
+        return ['/usr/bin/rsync', '-arRz', '--files-from=-',
```

Comment:
We might also want to use `--delete` flag

Reply:
I thought about it, but the problem is that we actually `rsync` files from multiple sources into the same target directory, so `--delete` would remove things we actually want to be there.

Reply:
This might be a problem when we remove files, and the remote device will have old copy of it.

Reply:
I think the only way we could use `--delete` would be to locally build the directory tree we want on the remote system first, then `rsync` that all at once. Maybe doing that is worthwhile; I might try it as an experiment to see how it looks.

Reply:
I should have added that the problem you're talking about is one we have now; it isn't a new problem introduced by this PR. The code does do an `rm -rf` under some circumstances (which I haven't changed); using `--delete` would mean we could get rid of that, which would be good I think.

Reply:
The thing that makes all of this awkward is that `lit` doesn't make a directory for each test; it only makes a directory for each directory full of tests, and then the tests get to decide what they do with `%t`. Some of them use it to generate one or more filenames that we have to copy by prefix, and some of them make a directory; there's nothing stopping them from doing both, either.

Without changing `lit` to always give a test its own directory, it's difficult to use the `--delete` flag; you'd have to build a copy of the directory hierarchy that you want on the remote on the local machine and then transfer that. That sounds easy enough in principle, but then there's the problem of what happens when you go to transfer the data back again — rsync would get it back to the built up hierarchy, but then you'd have to go through and copy things back to the original locations.

I think that's too big a change for this PR.

Reply:
I filed https://github.com/llvm/llvm-project/issues/53542 to add support for `lit` to create a per-test temporary directory (rather than, as presently, per suite). If that gets done, we could probably revisit the way all of this works as using `--delete` will be practical at that point.